src/software/new_geom/polynomial.cpp

```diff
@@ -0,0 +1,136 @@
+#include "software/new_geom/polynomial.h"
+
+#include <stdexcept>
+
+Polynomial::Polynomial() {}
+
+Polynomial::Polynomial(const std::vector<double> &coeffs) : coeffs(coeffs) {}
+
+Polynomial::Polynomial(const std::initializer_list<double> &coeffs) : coeffs(coeffs) {}
+
+Polynomial::Polynomial(const std::pair<double, double> &constraint1,
+                       const std::pair<double, double> &constraint2)
+{
+    if (constraint1.first == constraint2.first)
+    {
+        throw std::invalid_argument("Both inputs are equal - does not define a function");
+    }
+    double slope = (constraint2.second - constraint1.second) /
+                   (constraint2.first - constraint1.first);
+    coeffs.push_back(constraint1.second - (constraint1.first * slope));
+    coeffs.push_back(slope);
+}
+
+double Polynomial::getCoeff(unsigned int order) const
+{
+    return order >= coeffs.size() ? 0 : coeffs[order];
```

Comment:
nit; just did a bit of a double-take while reading this, could we change to:
``` suggestion
    return (order >= coeffs.size()) ? 0 : coeffs[order];
```

Reply:
Please don't use tertiary operators. Clarity > line count

Reply:
Done in af20cdc76f49a87c92abe851d79b90511c7bfb33, removed ternary operator